src/context/BookingState.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+import * as React from 'react';
+
+type Props = {
+  children: React.Node,
+};
+
+export type BookingStateType = {
+  bookingPage: 'SINGLE_BOOKING' | 'ALL_BOOKINGS',
+  selectedBooking: ?string,
+};
+
+export const ClickAllBooking = React.createContext(() => {});
+export const ClickSelectBooking = React.createContext(
+  (id: string) => {}, // eslint-disable-line no-unused-vars
+);
+const initBookingState = {
+  bookingPage: 'SINGLE_BOOKING',
+  selectedBooking: null,
+};
+export const BookingState = React.createContext();
+
+class BookingStateProvider extends React.Component<Props, BookingStateType> {
+  state = initBookingState;
+
+  onClickAllBooking = () => {
+    this.setState({ bookingPage: 'ALL_BOOKINGS' });
+  };
+
+  onClickSelect = (id: string) => {
+    this.setState({ bookingPage: 'SINGLE_BOOKING', selectedBooking: id });
+  };
+
+  render() {
+    const { bookingPage, selectedBooking } = this.state;
+
+    return (
+      <ClickAllBooking.Provider value={this.onClickAllBooking}>
```

Comment:
Do we really need three contexts here? What about renaming the main context to `BookingScreen` or sth. And Provide from that context both the screen info and the switchers?(see Line 41-42).
Eg.

![image](https://user-images.githubusercontent.com/2466887/40550032-4da181ca-6039-11e8-9c47-ecb486bc58a1.png)

Then we could consume from the required ones like `src/SingleBookingPage/bookingItem/Header.js` de-structuring like this:
![image](https://user-images.githubusercontent.com/2466887/40550139-927a4e80-6039-11e8-8dae-b084f696eb07.png)

Or even better, exporting a set of HOC from our `BookingScreen`(now BookingState+Clickers) context to decorate the consumer with the necessary props; just like we do with our User context now: `withLogin`, `withLogout`, etc.

Reply:
fixed